docs/proposals/container-runtime-interface/logging.md

```diff
@@ -0,0 +1,107 @@
+# Logging in Container Runtime Interface
+
+The following proposal summarizes the current problems encountered during
+cluster log aggregation in Kubernetes and proposes a solution to the problem on
+the container runtime level.
+
+## Motivation
+
+Logging is an essential part of cluster deployments, currently the cluster
+logging is facilitated via adhoc solutions and plugins. Use of different
+runtimes and heterogenity of logging format ecosystem makes it difficult to have
+a clean first-class solution to logging.
+
+In order to simplify the deployments and reduce the developer workload, it is
+imperative to provide a sane way to expose application logs and context
+metadata.
+
+## Solution
+
+The proposed solution is a volume-based approach and a loose logging format that
+will be specified using the container runtime interface to provide a coherent
+logging API.
+
+Kubernetes will have logging volumes for containers that will be used by
+runtimes. Each container will be assigned an append-only file on the logging
+volume of the pod. The logging volume lifecycle will be managed by Kubernetes,
+namely rotation and truncation, according to the logging policy specified by the
+user.
+
+It will be the container runtime's responsibility to redirect the logs of a container
+to the specified location, be it stdout/stderr logs or log files themselves.
+
+## Log Delimiting
+
+We cannot hope to dictate the logging format that will be used by the consumers.
```

Comment:
Yes, we need to enforce some logging format to support `kubectl log --since`, etc. I don't think we can completely avoid this.